configure.ac

```diff
@@ -39,6 +39,57 @@ AC_MSG_NOTICE([building without systemd])
 AM_CONDITIONAL(HAVE_SYSTEMD, false)
 fi
 
+AC_ARG_WITH(dnf,
+    AS_HELP_STRING([--with-dnf],[use dnf]))
+AC_ARG_WITH(yum,
+    AS_HELP_STRING([--with-yum],[use yum]))
+if test "$with_dnf" == yes
+then
+    AM_CONDITIONAL(USE_DNF, true)
+    if test "$with_yum" == yes
+    then
+        AM_CONDITIONAL(USE_YUM, true)
+    else
+        AM_CONDITIONAL(USE_YUM, false)
+    fi
+else
+if test "$with_yum" == yes
+then
+    AM_CONDITIONAL(USE_YUM, true)
+    if test "$with_dnf" == yes
+    then
+        AM_CONDITIONAL(USE_DNF, true)
+    else
+        AM_CONDITIONAL(USE_DNF, false)
+    fi
+else
+if test "$with_dnf" == no
+then
+    AM_CONDITIONAL(USE_DNF, false)
+    if test "$with_yum" == no
+    then
+        AM_CONDITIONAL(USE_YUM, false)
+    else
+        AM_CONDITIONAL(USE_YUM, true)
+    fi
+else
+if test "$with_yum" == no
+then
+    AM_CONDITIONAL(USE_YUM, false)
+    if test "$with_dnf" == no
+    then
+        AM_CONDITIONAL(USE_DNF, false)
+    else
+        AM_CONDITIONAL(USE_DNF, true)
+    fi
+else
+    AM_CONDITIONAL(USE_DNF, true)
+    AM_CONDITIONAL(USE_YUM, true)
+fi
+fi
+fi
+fi
+
```

Comment:
This looks too complicated to me. If I understand it right, this does the same as the following code.
```
AC_ARG_WITH(dnf,
AS_HELP_STRING([--with-dnf],[use dnf (default is YES)]),
FAF_PARSE_WITH([dnf]))
if test -z "$NO_DNF" # if not defined
then
AC_MSG_NOTICE([building with dnf])
AM_CONDITIONAL(USE_DNF, true)
else
AC_MSG_NOTICE([building without dnf])
AM_CONDITIONAL(USE_DNF, false)
fi

AC_ARG_WITH(yum,
AS_HELP_STRING([--with-yum],[use yum (default is YES)]),
FAF_PARSE_WITH([yum]))
if test -z "$NO_YUM" # if not defined
then
AC_MSG_NOTICE([building with yum])
AM_CONDITIONAL(USE_YUM, true)
else
AC_MSG_NOTICE([building without yum])
AM_CONDITIONAL(USE_YUM, false)
fi
```

Correct me if I am wrong.

Reply:
It is not quite the same, but the core of the logic is equal. I will update the code in my next commit according to you prosal. Thank you.